src/test/java/io/vertx/ext/asyncsql/SQLTestBase.java

```diff
@@ -0,0 +1,490 @@
+      conn = ar.result();
+      setupSimpleTable(conn, ar2 -> {
+        client.getConnection(ar3 -> {
+          ensureSuccess(context, ar3);
+          connections[0] = ar3.result();
+          connections[0].setAutoCommit(false, ar31 -> {
+            ensureSuccess(context, ar31);
+            connections[0].updateWithParams("UPDATE test_table SET name=? WHERE id=?",
+                new JsonArray().add(name).add(id), ar32 -> {
+                  ensureSuccess(context, ar32);
+                  context.assertEquals(ar32.result().getUpdated(), 1);
+                  connections[0].rollback(ar33 -> {
+                    ensureSuccess(context, ar33);
+                    connections[0].close(v -> {
+                      ensureSuccess(context, v);
+                      async1.complete();
+                    });
+                  });
+                });
+          });
+        });
+        client.getConnection(ar4 -> {
+          ensureSuccess(context, ar4);
+          connections[1] = ar4.result();
+          connections[1].query("SELECT name FROM test_table ORDER BY id", ar41 -> {
+            ensureSuccess(context, ar41);
+            ResultSet resultSet = ar41.result();
+            context.assertEquals(resultSet.getResults().get(0).getString(0), "Albert");
+            connections[1].close(v -> {
+              ensureSuccess(context, v);
+              async2.complete();
+            });
+          });
+        });
+      });
+    });
+  }
+
+  @Test
+  public void testSetAutocommitWhileInTransaction(TestContext context) {
+    //TODO not sur it is the correct translation, need to be checked
+    int id = 0;
+    String name = "Adele";
+    Async async = context.async();
+    client.getConnection(ar -> {
+      ensureSuccess(context, ar);
+      conn = ar.result();
+      setupSimpleTable(conn, ar2 -> {
+        ensureSuccess(context, ar2);
+        client.getConnection(ar3 -> {
+          ensureSuccess(context, ar3);
+          SQLConnection conn1 = ar3.result();
+          conn1.setAutoCommit(false, ar4 -> {
+            ensureSuccess(context, ar4);
+            conn1.updateWithParams("UPDATE test_table SET name=? WHERE id=?",
+                new JsonArray().add(name).add(id),
+                ar5 -> {
+                  ensureSuccess(context, ar5);
+                  conn1.setAutoCommit(true, ar6 -> {
+                    ensureSuccess(context, ar6);
+                    context.assertEquals(1, ar5.result().getUpdated());
+                    client.getConnection(ar7 -> {
+                      ensureSuccess(context, ar7);
+                      SQLConnection conn2 = ar7.result();
+                      conn2.query("SELECT name FROM test_table ORDER BY id", ar8 -> {
+                        ensureSuccess(context, ar8);
+                        context.assertEquals(ar8.result().getResults()
+                            .get(0).getString(0), name);
+                        conn2.close(v -> conn1.close(v2 -> async.complete()));
```

Comment:
Ahh, got it. `conn` is closed in the `@After` method anyways. No need to close it by hand then. :+1: